lib/DiscoveryStreamFeed.jsm

```diff
@@ -1286,10 +1287,90 @@ defaultLayoutResp = {
     {
       width: 12,
       components: [
+        {
+          type: "Message",
+          header: {
+            title: "Not Sponsored by Collection Title",
+            subtitle: "Not Sponsored by Brand Name",
+          },
+          properties: {},
+        },
+        {
+          type: "Message",
+          header: {
+            title: "Sponsored by Collection Title",
+            subtitle: "Sponsored by Brand Name",
+          },
+          campaign_id: "123456",
+          sponsored: true,
+          properties: {},
+        },
+      ],
+    },
+    {
+      width: 12,
+      campaign_id: "123456",
+      components: [
+        {
+          type: "Message",
+          header: {
+            title: "campaign but not paid",
+            subtitle: "campaign but not paid",
+          },
+          properties: {},
+        },
+        {
+          type: "Message",
+          header: {
+            title: "campaign but not paid",
+            subtitle: "campaign but not paid",
+          },
+          properties: {},
+        },
+      ],
+    },
+    {
+      width: 12,
+      components: [
+        {
+          type: "Message",
+          campaign_id: "123456",
```

Comment:
Is Message component has special handling by assigning campaign_id. Clarifying how campaign_id here is different from campaign_id in Line 1356 below

Reply:
Line 1356 is to make a whole collection dissmiss-able, and everything it wraps, line 1337 is to make just a single component dismiss-able.

Reply:
In the above example of a component being dismiss-able, and that component being a message component, we basically have a wrapper for a promo text component.